File: MorskoGram.Web.ViewModels/Posts/ListPostsViewModel.cs
namespace MorskoGram.Web.ViewModels.Posts
{
    using System;
    using MorskoGram.Data.Common.Models;
    using MorskoGram.Data.Models;
    using MorskoGram.Services.Mapping;

    public class ListPostsViewModel: BaseModel<Guid>, IMapFrom<Post>
    {
        public string Caption { get; set; }

        public string ImageLink { get; set; }

        public string CreatorEmail { get; set; }

        public int CommentsCount { get; set; }

        public int LikesCount { get; set; }
    }
}

File: MorskoGram.Web.API/ClientApp/src/models/posts/ListPostsViewModel.js
import {Mergeable} from '../Mergeable';

export class ListPostsViewModel extends Mergeable {
    constructor(obj) {
        super(obj);

        if (!obj) {
            this.id = '';
            this.imageLink = '';
            this.creatorEmail = '';
            this.createdOn = '';
            this.caption = '';
            this.likesCount = 0;
            this.commentsCount = 0;
        }
    }
}
